### PR TITLE
Leetcose 125 solution

### PR DESCRIPTION
**Pull request by silulyu**

### CODE REVIEW

File: L125_Valid_Palindrome.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Input
+# a string
```
Comment (shuoli7) [inline L125_Valid_Palindrome.py]: s/a string/A string

File: L125_Valid_Palindrome.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Input
+# a string 
+
+# Output
+# determine if it is a palindrom
```
Comment (shuoli7) [inline L125_Valid_Palindrome.py]: s/determine/Determine
s/palindrom/palindrome

Comment (shuoli7) [reply L125_Valid_Palindrome.py]: Output should be
A boolean true or false to determine if the input string is a palindrome.

File: L125_Valid_Palindrome.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+# Solution
+# To solve the problem, I use two pointers to check 
+# whether the leftmost character is equal to the rightmost character
+# During the comparison, if the two character is not the same, stop and return False
+# If not, the comparison continues until the two pointers meet.
```
Comment (shuoli7) [inline L125_Valid_Palindrome.py]: If two characters are same

Comment (shuoli7) [reply L125_Valid_Palindrome.py]: Give more details how to proceed to next round check like
If two characters are same, then we increment left pointer and decrement right pointer, until two characters are different or pointers meet.

At the end, if return false condition is not met, then we return true.

File: L125_Valid_Palindrome.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+# If the string is empty, is it a valid palindrome?
+
+class Solution(object):
+    def isPalindrome(self, s):
+        if not s:
+            return True
+        i = 0
+        j = len(s) - 1
+        while i < j:
+            while i < j and not s[i].isalnum():
```
Comment (shuoli7) [inline L125_Valid_Palindrome.py]: Give some inline comments for this check. The goal is to filter out non alphabetical and non numerical characters. You may ask if the given string has char other than alphabetical char and number or you can use certain function to filter them out at the beginning.

For example:
>>> import re
>>> your_string = re.sub("[^a-zA-Z0-9]", "", your_string)
^ is not, sub is substitute, [^a-zA-Z0-9] is called regular expression, regex.

This means replace anything other than A-Z, a-z, 0-9 to "".

File: L125_Valid_Palindrome.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+                j -= 1
+            if s[i].lower() != s[j].lower():
+                return False
+            i += 1
+            j -= 1
+        return True
+     
+#Time Complexity: O(n)
+#Space Complexity: O(1)
+
```
Comment (shuoli7) [inline L125_Valid_Palindrome.py]: Remove 36, 37 empty lines.

File: L125_Valid_Palindrome.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+            while i < j and not s[j].isalnum():
+                j -= 1
+            if s[i].lower() != s[j].lower():
+                return False
+            i += 1
+            j -= 1
+        return True
+     
+#Time Complexity: O(n)
+#Space Complexity: O(1)
```
Comment (shuoli7) [inline L125_Valid_Palindrome.py]: Space after #

File: L125_Valid_Palindrome.py (hunk 1 above)
Comment (silulyu) [reply L125_Valid_Palindrome.py]: fixed

File: L125_Valid_Palindrome.py (hunk 2 above)
Comment (silulyu) [reply L125_Valid_Palindrome.py]: fixed

File: L125_Valid_Palindrome.py (hunk 4 above)
Comment (silulyu) [reply L125_Valid_Palindrome.py]: fixed

File: L125_Valid_Palindrome.py (hunk 3 above)
Comment (silulyu) [reply L125_Valid_Palindrome.py]: fixed

File: L125_Valid_Palindrome.py (hunk 5 above)
Comment (silulyu) [reply L125_Valid_Palindrome.py]: fixed

File: L125_Valid_Palindrome.py (hunk 6 above)
Comment (silulyu) [reply L125_Valid_Palindrome.py]: fixed

File: L125_Valid_Palindrome.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
+# Input
+# A string 
+
+# Output
+# A boolean true or false to determine if the input string is a palindrome.
+
+# Solution
+# To solve the problem, I use two pointers to check 
+# whether the leftmost character is equal to the rightmost character
+# During the comparison, if the two character is not the same, stop and return False
```
Comment (shuoli7) [inline L125_Valid_Palindrome.py]: Empty line between 9 and 10.

File: L125_Valid_Palindrome.py [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+# If two characters are same, we increment left pointer and decrement right pointer 
+# until two characters are different or pointers meet. 
+# At the end, if return false condition is not met, we return true.
```
Comment (shuoli7) [inline L125_Valid_Palindrome.py]: Empty line between 12 and 13.

File: L125_Valid_Palindrome.py [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+# Clarification
+# If the string is empty, is it a valid palindrome?
```
Comment (shuoli7) [inline L125_Valid_Palindrome.py]: Add another clarification question:
Does the input string contain non-alphanumeric chars?

File: L125_Valid_Palindrome.py [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+# Solution
+# To solve the problem, I use two pointers to check 
+# whether the leftmost character is equal to the rightmost character
+# During the comparison, if the two character is not the same, stop and return False
+# If two characters are same, we increment left pointer and decrement right pointer 
+# until two characters are different or pointers meet. 
+# At the end, if return false condition is not met, we return true.
+
+# Clarification
```
Comment (shuoli7) [inline L125_Valid_Palindrome.py]: Actually, always move Clarification question on the top, before Input.

Comment (silulyu) [reply L125_Valid_Palindrome.py]: I feel it will be better after the input and output. What do you think?

**Pull request merged**